R/sample_predictions.R

```diff
@@ -163,30 +181,40 @@ combine_obs_with_pred <- function(predicted_counts,
 #'   reporting_triangle,
 #'   disp,
 #'   500,
-#'   fun_to_aggregate = sum,
-#'   k = 2
+#'   ref_time_aggregator = function(x) zoo::rollsum(x, k = 2, align = "right")
 #' )
 #' nowcast_pred_draws_rolling_df
-sample_predictions <- function(point_nowcast_matrix,
-                               reporting_triangle,
-                               dispersion,
-                               draws = 1000,
-                               fun_to_aggregate = sum,
-                               k = 1) {
+sample_predictions <- function(
+    point_nowcast_matrix,
+    reporting_triangle,
+    uncertainty_params,
+    draws = 1000,
+    error_model = function(pred, params) sample_nb(pred, params),
+    ref_time_aggregator = function(x) identity(x),
```

Comment:
As these are now  matrix operations are we confident ref_time and delay is the right name? They are actually linked to when they occur within the wider operation right? I don't have concise thoughts here but worth a review I think.